### PR TITLE
[CARBONDATA-2418] [Presto] [S3] Fixed Presto Can't Query CarbonData When CarbonStore is at S3

### PR DESCRIPTION
**Pull request by anubhav100**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 
 - [x] Any backward compatibility impacted?
 
 - [x] Document update required?

 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4608/

**Comment by anubhav100** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5767/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4826/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4616/

File: integration/presto/README.md [hunk 1]
```diff
@@ -82,6 +82,29 @@ Please follow the below steps to query carbondata in presto
   For example, if you have a schema named 'default' stored in hdfs://namenode:9000/test/carbondata/,
   Then set carbondata-store=hdfs://namenode:9000/test/carbondata
   
+#### Connecting to carbondata store on s3
+ * In case you want to query carbonstore on S3 using S3A api put following additional properties inside $PRESTO_HOME$/etc/catalog/carbondata.properties 
+   ```
+    fs.s3a.access.key={value}
+    fs.s3a.secret.key={value}
+    Optional: fs.s3a.endpoint={value}
```
Comment (xuchuanyin) [inline integration/presto/README.md]: better to describe the properties as below to make it convenient for user to copy&paste.
```
# Required properties
A=B
# Optional properties
C=D
```

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4764/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5918/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4949/

**Comment by anubhav100** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5926/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4771/

File: integration/presto/README.md (hunk 1 above)
Comment (anubhav100) [reply integration/presto/README.md]: done

**Comment by jackylk** [comment]
please remove datamap.mdtfile file in the commit
@anubhav100

File: integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableConfig.java [hunk 2]
```diff
@@ -51,7 +59,8 @@
     return this;
   }
 
-  @NotNull public String getStorePath() {
+  @NotNull public String
+  getStorePath() {
```
Comment (jackylk) [inline integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableConfig.java]: no need to change

File: integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableConfig.java [hunk 3]
```diff
@@ -60,6 +69,7 @@
     return this;
   }
 
+
```
Comment (jackylk) [inline integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableConfig.java]: remove empty line

**Comment by anubhav100** [comment]
@jackylk i have done changes please review

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5156/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6182/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5021/

**Comment by anubhav100** [comment]
@xuchuanyin @jackylk please review

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**